tests/infra_test.go

```diff
@@ -221,19 +221,24 @@ var _ = Describe("Infrastructure", func() {
 			})
 
 			AfterEach(func() {
-				By("removing the taint from the tainted node")
-				var taints []k8sv1.Taint
-
-				for _, taint := range selectedNode.Spec.Taints {
-					if taint.Key != "CriticalAddonsOnly" {
-						taints = append(taints, taint)
+				if selectedNode != nil {
+					defer func() {
```

Comment:
when does the defer triggered in tests? is it after each test (i.e. at the end of every `It`)?

Reply:
I think i know the answer - is it after the context is closed. in this case it'll be in the end of the `AfterEach`, right?

Reply:
yes, its in AfterEach , just in case selectedNode is not nil (this what the deferred func does, just nil it, in case of exceptions)

Reply:
[yep](https://tour.golang.org/flowcontrol/12)

Reply:
i might need to remove it, in case it doesnt change the ptr as in C (passed by val versus passed by ref)

Reply:
removed, we also already have
```
			BeforeEach(func() {
				selectedNode = nil
			})
```